haystack_experimental/tools/human_in_the_loop.py

```diff
@@ -0,0 +1,183 @@
+        """
+        Ask for user confirmation before executing a tool.
+
+        :param tool_name: Name of the tool to be executed.
+        :param params: Parameters to be passed to the tool.
+        :returns:
+            ConfirmationResult with action (e.g. "confirm" or "reject"), optional feedback message and new parameters
+            if modified.
+        """
+        # Display info
+        lines = [f"[bold yellow]Tool:[/bold yellow] {tool_name}"]
+        if params:
+            lines.append("\n[bold yellow]Arguments:[/bold yellow]")
+            for k, v in params.items():
+                lines.append(f"\n[cyan]{k}:[/cyan]\n  {v}")
+        self.console.print(Panel("\n".join(lines), title="🔧 Tool Execution Request"))
+
+        # Ask action
+        choice = Prompt.ask(
+            "\nYour choice",
+            choices=["y", "n", "m"],  # confirm, reject, modify
+            default="y",
+        )
+        if choice == "y":
+            return ConfirmationResult(action="confirm")
+        elif choice == "m":
+            new_params = {}
+            for k, v in params.items():
+                new_val = Prompt.ask(f"Modify '{k}'", default=str(v))
+                new_params[k] = new_val
+            return ConfirmationResult(action="modify", new_params=new_params)
+        else:  # reject
+            feedback = Prompt.ask("Feedback message (optional)", default="")
+            return ConfirmationResult(action="reject", feedback=feedback or None)
+
+
+class SimpleInputPrompt:
+    """
+    Simple confirmation prompt using standard input/output.
+    """
+
+    def confirm(self, tool_name: str, params: dict[str, Any]) -> ConfirmationResult:
+        """
+        Ask for user confirmation before executing a tool.
+
+        :param tool_name: Name of the tool to be executed.
+        :param params: Parameters to be passed to the tool.
+        :returns:
+            ConfirmationResult with action (e.g. "confirm" or "reject"), optional feedback message and new parameters
+            if modified.
+        """
+        print(f"Tool: {tool_name}")
+        if params:
+            print("Arguments:")
+            for k, v in params.items():
+                print(f"  {k}: {v}")
+
+        choice = input("Confirm execution? (y=confirm / n=reject / m=modify): ").strip().lower()
+        if choice == "y":
+            return ConfirmationResult(action="confirm")
+        elif choice == "m":
+            new_params = {}
+            for k, v in params.items():
+                new_val = input(f"Modify '{k}' [{v}]: ").strip() or v
+                new_params[k] = new_val
+            return ConfirmationResult(action="modify", new_params=new_params)
+        else:  # modify
+            feedback = input("Feedback message (optional): ").strip()
+            return ConfirmationResult(action="reject", feedback=feedback or None)
+
+
+class DefaultPolicy:
+    """
+    Default execution policy:
```

Comment:
When I read policy, I first thought this is a choice between AlwaysAsk, AskOnce, NeverAsk. Here, the user is always asked.